cli/pcluster/resources/custom_resources/custom_resources_code/cleanup_resources.py

```diff
@@ -20,6 +20,31 @@
 boto3_config = Config(retries={"max_attempts": 60})
 
 
+def _delete_dns_records(event):
+    """Delete all DNS entries from the private Route53 hosted zone created within the cluster."""
+    hosted_zone_id = event["ResourceProperties"]["ClusterHostedZone"]
+    logger.info("Deleting DNS records from %s", hosted_zone_id)
+
+    if hosted_zone_id:
+        try:
+            route53 = boto3.client("route53")
+            pagination_config = {"PageSize": 100}
+            paginator = route53.get_paginator("list_resource_record_sets")
+            for page in paginator.paginate(HostedZoneId=hosted_zone_id, PaginationConfig=pagination_config):
```

Comment:
to be sure everything gets deleted (since we cannot retry this lambda function) we could add a while on top of the for that checks again if there are no remaining record sets. By doing this we can avoid failing on exceptions in the inner loop. Something similar to what's done in `_terminate_cluster_nodes`